test/services/datasources/datasources.test.js

```diff
@@ -26,49 +26,90 @@ describe('datasources service', () => {
 		expect(result.config).to.exist;
-		expect(result.config).to.haveOwnProperty('type');
+		expect(result.config).to.haveOwnProperty('target');
 		expect(result.name).to.exist;
 		expect(result.createdBy.toString()).to.equal(admin._id.toString());
 		expect(result.schoolId.toString()).to.equal(admin.schoolId.toString());
 		datasourceModel.deleteOne({ _id: result._id }).lean().exec();
 	});
 
+	it('protects fields', async () => {
+		const admin = await testObjects.createTestUser({ roles: ['administrator'] });
+		const params = await generateRequestParamsFromUser(admin);
+		const data = {
+			config: { target: 'csv', csvpassword: 'secure123' },
+			name: `test${Date.now()}`,
+			protected: ['csvpassword'],
+		};
+		const result = await datasourcesService.create(data, params);
+		expect(result).to.not.be.undefined;
+		expect(result.config).to.exist;
+		expect(result.config).to.haveOwnProperty('target');
+		expect(result.name).to.exist;
+		expect(result.createdBy.toString()).to.equal(admin._id.toString());
+		expect(result.schoolId.toString()).to.equal(admin.schoolId.toString());
+		expect(result.protected).to.not.be.undefined;
+		datasourceModel.deleteOne({ _id: result._id }).lean().exec();
+	});
+
 	it('GET a datasource', async () => {
 		const admin = await testObjects.createTestUser({ roles: ['administrator'] });
 		const params = await generateRequestParamsFromUser(admin);
 		const data = {
-			config: { type: 'csv' },
+			config: { target: 'csv' },
 			name: `test${Date.now()}`,
 		};
 		const datasource = await datasourcesService.create(data, params);
 		params.query = {};
 		const result = await datasourcesService.get(datasource._id, params);
 		expect(result).to.not.be.undefined;
 		expect(result.config).to.exist;
-		expect(result.config).to.haveOwnProperty('type');
+		expect(result.config).to.haveOwnProperty('target');
 		expect(result.name).to.exist;
 		expect(result.createdBy.toString()).to.equal(admin._id.toString());
 		expect(result.schoolId.toString()).to.equal(admin.schoolId.toString());
 		datasourceModel.deleteOne({ _id: result._id }).lean().exec();
 	});
 
+	it('protected fields are protected', async () => {
+		const admin = await testObjects.createTestUser({ roles: ['administrator'] });
+		const params = await generateRequestParamsFromUser(admin);
+		const data = {
+			config: {
+				target: 'csv', password: 'didumm', secret: 'Im an agent', public: 'im an expert',
+			},
+			name: `test${Date.now()}`,
+			protected: ['password', 'secret'],
+		};
+		const datasource = await datasourcesService.create(data, params);
+		params.query = {};
+		const result = await datasourcesService.get(datasource._id, params);
+		expect(result).to.not.be.undefined;
+		expect(result.config).to.exist;
+		expect(result.config.password).to.equal('secret');
+		expect(result.config.secret).to.equal('secret');
+		expect(result.config.public).to.equal('im an expert');
+		datasourceModel.deleteOne({ _id: result._id }).lean().exec();
```

Comment:
It is slowed the test process if every task delete, the created data. Maybe add a helper to testObjects and handle it over after/afterAll with cleanup() -> datasourceModel.delete( <list of ids>)?

Reply:
Can you create a ticket, that we track this task?